src/Symfony/Component/Routing/Route.php

```diff
@@ -502,30 +525,40 @@ public function getCondition()
     public function setCondition(?string $condition)
     {
         $this->condition = (string) $condition;
-        $this->compiled = null;
+        $this->matcherCompiled = null;
+        $this->generatorCompiled = null;
 
         return $this;
     }
 
     /**
      * Compiles the route.
      *
+     * @param bool $forGenerator If the route should be compiler for the generator
      * @return CompiledRoute A CompiledRoute instance
      *
      * @throws \LogicException If the Route cannot be compiled because the
      *                         path or host pattern is invalid
      *
      * @see RouteCompiler which is responsible for the compilation process
      */
-    public function compile()
+    public function compile(bool $forGenerator = false)
```

Comment:
I'm not sure to get why we need a different compilation result for generator vs dumper?
If we really need this, then I would prefer to deal with the optimization inside the dumper, so that no public API needs to change.

Reply:
Maybe it's unclear but I meant compiling differently for matching vs generation.
Generation shouldn't be modified because of strictRequirements and stuff. Therefore in fact I made a mistake, `$forGeneration` should instead be `$forMatching`.
I would rather as well keeping implementation inside the dumper.
I'll try another approach thanks